figshare/Article.py

```diff
@@ -758,33 +765,63 @@ def __can_copy_files(self, version_data):
+            self.logs.write_log_in_file("info", "Copying curation UAL_RDM files to preservation UAL_RDM folder.", True)
+            success = success & self.__copy_files_ual_rdm(version_data, folder_name)
+            # save json in metadata folder for each version
+            self.logs.write_log_in_file("info", "Saving json in metadata folder for each version.", True)
+            success = success & self.__save_json_in_metadata(version_data, folder_name)
+
+        if check_files and copy_files:
             # download all files and verify hash with downloaded file.
             delete_now = self.__download_files(version_data['files'], version_data, folder_name)
             # check if download process has error or not.
             if (delete_now is False):
                 # copy curation UAL_RDM files in storage UAL_RDM folder for each version
                 self.logs.write_log_in_file("info", "Copying curation UAL_RDM files to preservation UAL_RDM folder for each version.", True)
-                self.__copy_files_ual_rdm(version_data, folder_name)
+                success = success & self.__copy_files_ual_rdm(version_data, folder_name)
                 # check and create empty directories for each version
                 self.logs.write_log_in_file("info", "Checking and creating empty directories for each version.", True)
-                self.create_required_folders(version_data, folder_name)
+                success = success & self.create_required_folders(version_data, folder_name)
                 # save json in metadata folder for each version
                 self.logs.write_log_in_file("info", "Saving json in metadata folder for each version.", True)
-                self.__save_json_in_metadata(version_data, folder_name)
-                value_post_process = self.processor.post_process_script_function("Article", check_dir, value_pre_process)
-                if (value_post_process != 0):
-                    self.logs.write_log_in_file("error", f"{version_data['id']} version {version_data['version']} - Post-processing script failed.",
+                success = success & self.__save_json_in_metadata(version_data, folder_name)
+
+                # only run the postprocessor if all above steps succeeded
+                if success:
+                    value_post_process = self.processor.post_process_script_function("Article", check_dir, value_pre_process)
+                    if (value_post_process != 0):
+                        self.logs.write_log_in_file("error",
+                                                    f"{version_data['id']} version {version_data['version']} - Post-processing script failed.",
+                                                    True)
+                else:
+                    self.logs.write_log_in_file("info",
+                                                f"No further processing for {version_data['id']} version {version_data['version']} due to errors.",
```

Comment:
success = false here.  Shall this boolean "success" be reset?

Reply:
Since there are no loops in this function, success is already reset to true every time the function is called (L768).  Furthermore, if the code reaches the statement you've indicated, the function will exit immediately after since none of the other else statements below it will be entered. All that is to say, resetting it here won't have any effect (that I can see)